src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4182,6 +4377,80 @@ protected override void VisitArguments(ImmutableArray<BoundExpression> arguments
             return (method, results, shouldReturnNotNull);
         }
 
+        private void ApplyMemberPostConditions(MethodSymbol method)
+        {
+            if (method is null)
+            {
+                return;
+            }
+
+            do
+            {
+                var type = method.ContainingType;
+                var notNullMembers = method.NotNullMembers;
+                var notNullWhenTrueMembers = method.NotNullWhenTrueMembers;
+                var notNullWhenFalseMembers = method.NotNullWhenFalseMembers;
+
+                if (notNullWhenTrueMembers.IsEmpty && notNullWhenFalseMembers.IsEmpty)
+                {
+                    // TODO2 apply in conditional state too?
+                    if (IsConditionalState)
+                    {
+                        applyMemberPostConditions(type, notNullMembers, ref StateWhenTrue);
+                        applyMemberPostConditions(type, notNullMembers, ref StateWhenFalse);
+                    }
+                    else
+                    {
+                        applyMemberPostConditions(type, notNullMembers, ref State);
+                    }
+                }
+                else
+                {
+                    Split();
+                    applyMemberPostConditions(type, notNullWhenTrueMembers, ref StateWhenTrue);
+                    applyMemberPostConditions(type, notNullWhenFalseMembers, ref StateWhenFalse);
+                }
+
+                method = method.OverriddenMethod;
```

Comment:
>method [](start = 25, length = 6)

Can you please identify the spec or LDM discussion that relates to this?

Reply:
See email thread about MemberNotNull design (section on overrides copied in description)